index.js

```diff
@@ -0,0 +1,163 @@
+(function() {
+  var display = {
+    current: 0,
+    first: 0,
+    second: 0,
+    operand: '',
+    result: '',
+    firstValueAdded: false,
+    secondValueAdded: false
+  }
+
+  var addTheListeners = function(){
+    var calculator = document.querySelector('.calculator-body')
+    calculator.addEventListener('click', function(event){
+      if(event.target.matches('.calculator-button:not(#flip), #decimal')){
+        displayValue(event.target.innerHTML)
+      }else if(event.target.matches('#divide, #multiply, #subtract, #add')){
+        getOperand(event.target.innerHTML)
+      }else if(event.target.matches('#AC')){
+        allClear()
+      }else if(event.target.matches('#percentage')){
+        percent()
+      }else if(event.target.matches('#equals')){
+        calculate()
+      }else if(event.target.matches('#flip')){
+        flipSign()
+      }
+    })
+  }
+  addTheListeners()
+
+  var changeFontSize = function(){
+    var maxLength = document.getElementsByClassName('calculator-valueDisplay')[0]
+    var currentLength = document.getElementsByClassName('calculator-valueDisplay')[0].innerHTML
+    if(currentLength.length >= maxLength.attributes.maxlength.value - 2){
+      maxLength.style.fontSize = "30px"
+    }
+  }
+  var displayValue = function(num){
+    if(display.secondValueAdded){
+      display.firstValueAdded = false
+      display.secondValueAdded = false
+      display.first = display.result
+      display.second = 0
+      display.current[0].innerHTML = 0
+    }
+    changeFontSize()
+    if(!display.firstValueAdded){
+      display.current = document.getElementsByClassName('calculator-valueDisplay')
+      if(display.current[0].innerHTML == 0){
+        display.current[0].innerHTML = null
+      }
+      display.current[0].innerHTML += num
+    }else{
+      if(display.first == 0){
+        display.first = display.current[0].innerHTML
+        display.current[0].innerHTML = 0
+      }
+      display.second = document.getElementsByClassName('calculator-valueDisplay')
+      if(display.current[0].innerHTML == 0){
+        display.current[0].innerHTML = null
+      }
+      display.current[0].innerHTML += num
+    }
+  }
+
+  var getOperand = function(symbol){
+    if(display.operand !== ''){
+      display.operand = ''
+      display.first = 0
+    }
+    switch(symbol){
+      case '+':
+        calculate()
```

Comment:
i cant seem to find the `calculate()` function definition. Is this because its a WIP?